onnx/shape_inference/implementation.cc

```diff
@@ -446,9 +449,9 @@ static void InferShapesImpl(
             *iter->second = *inferredType;
           }
         }
-        // if enable symbolic, check existing symbolic shape and generate new ssymbolic shape
-        SymbolicShape symbolicShape(g);
-        inferredType = enableSymbolicShape(inferredType, symbolicShape);
+        // Check existing symbolic shape and generate new symbolic shape
+        SymbolTable symbolTable(g);
```

Comment:
minor nit: recomputing the symbolTable every time seems less efficient; we need to create it just once.

Reply:
Good point. Now single InferShapes will only have single symbolTable (among InferShapesImpl, InferShapeForFunctionNode, doInferencing). To make subgraph inference share the same symbolTable from the main graph, I introduced symbolTable in graphInferenceContext. Thanks for the suggestion!